File: capavista/nuevoProducto.vb
Public Class nuevoProducto
    Private Sub ProductosBindingNavigatorSaveItem_Click(sender As Object, e As EventArgs)
        Me.Validate()
        Me.ProductosBindingSource.EndEdit()
        Me.TableAdapterManager.UpdateAll(Me.KioscoDataSet)

    End Sub

    Private Sub NuevoProducto_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        'TODO: esta línea de código carga datos en la tabla 'KioscoDataSet.tipoProducto' Puede moverla o quitarla según sea necesario.
        Me.TipoProductoTableAdapter.Fill(Me.KioscoDataSet.tipoProducto)
        'TODO: esta línea de código carga datos en la tabla 'KioscoDataSet.tipoProducto' Puede moverla o quitarla según sea necesario.
        Me.TipoProductoTableAdapter.Fill(Me.KioscoDataSet.tipoProducto)
        'TODO: esta línea de código carga datos en la tabla 'KioscoDataSet.productos' Puede moverla o quitarla según sea necesario.
        Me.ProductosTableAdapter.Fill(Me.KioscoDataSet.productos)

    End Sub

    Private Sub Btnagregar_Click(sender As Object, e As EventArgs) Handles btnagregar.Click

    End Sub
End Class

File: capavista/nuevoProducto.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using capalnegocio;
namespace capavista
{
    public partial class nuevoProducto : Form
    {
        private lnproducto productoLN = new lnproducto();
        private DataTable tabla = new DataTable();


        public nuevoProducto()
        {
            InitializeComponent();
        }



        private void Button1_Click(object sender, EventArgs e)
        {


            
            int stock, stockmin;
            String descripcion, tproducto;
            Double precio;
            stock = Convert.ToInt32(txtstock.Text);
            stockmin = Convert.ToInt32(txtmin.Text);
            precio = Convert.ToDouble(txtprecio.Text);
            descripcion = txtdescripcion.Text;
            tproducto = combotproducto.Text;
            

            
                if ((txtdescripcion.Text != "" && precio > 0.00) && (stock > 0 && stockmin > 0))
                {

                   
                   
                        productoLN.altaProducto(tproducto, descripcion, precio, stock, stockmin, dateTimePicker1.Value, dateTimePicker2.Value);
                        MessageBox.Show("Producto agregado correctamente");
                        dataGridView1.Rows.Add(descripcion);
                                        
                }


            
        }

        private void NuevoProducto_Load(object sender, EventArgs e)
        {
            // TODO: esta línea de código carga datos en la tabla 'kioscoDataSet.tipoProducto' Puede moverla o quitarla según sea necesario.
            this.tipoProductoTableAdapter.Fill(this.kioscoDataSet.tipoProducto);

            dateTimePicker1.Value = System.DateTime.Now;
            dateTimePicker2.Value = System.DateTime.Now;

        }
    }
}
